test/TestCaseReader.cpp

```diff
@@ -118,6 +118,7 @@ pair<map<string, string>, size_t> TestCaseReader::parseSourcesAndSettingsWithLin
 		{
 			if (boost::algorithm::starts_with(line, sourceDelimiterStart) && boost::algorithm::ends_with(line, sourceDelimiterEnd))
 			{
+				// If statement registers previously parsed source
 				if (!(currentSourceName.empty() && currentSource.empty()))
 					sources[currentSourceName] = std::move(currentSource);
```

Comment:
I'd drop this comment. There are so many `if`s here that it actually made it more confusing to me :)

Or, if you really think it's necessary, I'd reword it a bit and move it inside. I'd also move the negation inside parenthesis since this is the part that actually confused me the most (I did not notice the parenthesis and read it as `!A.empty() && B.empty()`).
```suggestion
				if (!currentSourceName.empty() || !currentSource.empty())
					// Finalize the previous file before starting a new one
					sources[currentSourceName] = std::move(currentSource);
```

Reply:
The code looks unchanged. Maybe you forgot to push this change?

Reply:
I forgot to drop the comment. Will push an update.

Reply:
Could you also change the `if` to `if (!currentSourceName.empty() || !currentSource.empty())`?